sw/device/lib/dif/dif_padctrl.h

```diff
@@ -0,0 +1,157 @@
+ * A pad can be an MIO or DIO pad
+ */
+typedef enum dif_padctrl_pad_kind {
+  kDifPadctrlPadMio = 0, /**< MIO (Multiplexed Input/Output) pad. Connected via
+                            pinmux. */
+  kDifPadctrlPadDio,     /**< DIO (Dedicated Input/Output) pad. Connected
+                            directly to peripheral inputs and outputs, bypassing
+                            pinmux. */
+} dif_padctrl_pad_kind_t;
+
+/**
+ * Pad index.
+ *
+ * Combined with `dif_padctrl_pad_kind_t` to refer to a specific pad.
+ */
+typedef uint32_t dif_padctrl_pad_index_t;
+
+/**
+ * Padctrl instance state.
+ *
+ * Padctrl persistent data that is required by the Padctrl API.
+ */
+typedef struct dif_padctrl {
+  mmio_region_t base_addr; /**< Padctrl base address. */
+} dif_padctrl_t;
+
+/**
+ * Padctrl init routine error codes.
+ */
+typedef enum dif_padctrl_init_result {
+  kDifPadctrlInitOk = kDifPadctrlOk,
+  kDifPadctrlInitError = kDifPadctrlError,
+  kDifPadctrlInitBadArg = kDifPadctrlBadArg,
+  kDifPadctrlInitLocked, /**< Peripheral is locked and cannot be reset or
+                            initialised. This is not recoverable. */
+} dif_padctrl_init_result_t;
+
+/**
+ * Initialises an instance of Padctrl.
+ *
+ * Information that must be retained, and is required to program Padctrl, shall
+ * be stored in @p padctrl.
+ *
+ * @param base_addr Base address of an isntance of the Padctrl IP block
+ * @param padctrl Padctrl state data
+ * @return dif_padctrl_init_result_t.
+ */
+dif_padctrl_init_result_t dif_padctrl_init(mmio_region_t base_addr,
+                                           dif_padctrl_t *padctrl);
+
+/**
+ * Locks the padctrl registers.
+ *
+ * After the registers have been locked, they can only be unlocked by the
+ * system reset.
+ *
+ * @param padctrl Padctrl state data.
+ * @return dif_padctrl_result_t.
+ */
+dif_padctrl_result_t dif_padctrl_lock(dif_padctrl_t *padctrl);
+
+/**
+ * Padctrl attribute enable routines error codes.
+ */
+typedef enum dif_padctrl_enable_attr_result {
+  kDifPadctrlEnableOk = kDifPadctrlOk,
+  kDifPadctrlEnableError = kDifPadctrlError,
+  kDifPadctrlEnableBadArg = kDifPadctrlBadArg,
+  kDifPadctrlEnableAttrNotSupported, /**< Attribute is not supported. A write
+                                        has been performed to the attribute
+                                        registers but it had no effect as the
+                                        attribute isn't supported. Enable can
+                                        be retried with a different
+                                        attribute. */
+  kDifPadctrlEnableAttrConflict,     /**< Could not enable attribute because it
+                                        conflicts with another already enabled
+                                        attribute. No write to the attribute
+                                        registers has been attemped. Enable can
+                                        be retried with a different attribute. */
```

Comment:
This looks kind of strange. For multi-line comments, I think it's best to write them like this:
```
/**
 * Blah Blah
 */
kMyConstant,
```
The inline notation can get a bit cramped (so I avoid it altogether).